modules/objdetect/src/aruco/aruco_detector.cpp

```diff
@@ -847,15 +665,232 @@ struct ArucoDetector::ArucoDetectorImpl {
+        std::sort(candidateTree.begin(), candidateTree.end());
+        // group index for each candidate
+        vector<int> groupId(candidateTree.size(), -1);
+        vector<vector<unsigned int> > groupedCandidates;
+        vector<bool> isSelectedContours(candidateTree.size(), true);
+        const float moduleGroupRate = (float)dictionary.markerSize*.5f + 1.f;
+                                      // (float)detectorParams.minMarkerDistanceRate*4.f*
+                                      // (dictionary.markerSize+2*detectorParams.markerBorderBits); // TODO: optimize parameter
+        const float moduleRate = .21f; // TODO: optimize parameter?
+
+        int countSelectedContours = 0;
+        for (unsigned int i = 0; i < candidateTree.size(); i++) {
+            for (unsigned int j = i + 1; j < candidateTree.size(); j++) {
+                float minDist = getDistance(candidateTree[i].corners, candidateTree[j].corners);
+                    // if mean square distance is too low, remov0e the smaller one of the two markers
+                    if(minDist < candidateTree[j].moduleSize*moduleGroupRate) {
+                        isSelectedContours[i] = false;
+                        isSelectedContours[j] = false;
+                        // i and j are not related to a group
+                        if(groupId[i] < 0 && groupId[j] < 0){
+                            // mark candidates with their corresponding group number
+                            groupId[i] = groupId[j] = (int)groupedCandidates.size();
+
+                            // create group
+                            vector<unsigned int> grouped;
+                            grouped.push_back(i);
+                            grouped.push_back(j);
+                            groupedCandidates.push_back(grouped);
+                        }
+                        // i is related to a group
+                        else if(groupId[i] > -1 && groupId[j] == -1) {
+                            int group = groupId[i];
+                            groupId[j] = group;
+
+                            // add to group
+                            groupedCandidates[group].push_back(j);
+                        }
+                        // j is related to a group
+                        else if(groupId[j] > -1 && groupId[i] == -1) {
+                            int group = groupId[j];
+                            groupId[i] = group;
+
+                            // add to group
+                            groupedCandidates[group].push_back(i);
+                        }
+                    }
+            }
+            countSelectedContours += isSelectedContours[i];
+        }
+        for (vector<unsigned int>& grouped : groupedCandidates) {
+            std::sort(grouped.begin(), grouped.end());
+            if (detectorParams.detectInvertedMarker) // if detectInvertedMarker choose smallest contours
+                std::reverse(grouped.begin(), grouped.end());
```

Comment:
I propose to use different sort calls depending on `detectorParams.detectInvertedMarker`. It allows to get rif of std::revrese